module-1/src/Components/MovieForm/MovieForm.tsx

```diff
@@ -1,16 +1,116 @@
+import { ChangeEvent, useState } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { toast } from "react-toastify";
+import { TestData } from "../../TestData";
 import "./MovieForm.css";
 interface MovieFormProps {
   title: string;
   movie?: Movie;
   closeCallback: () => void;
 }
 
+type FormData = {
+  id: number;
+  title: string;
+  tagline?: string;
+  vote_average: number;
+  vote_count?: number;
+  release_date: string;
+  poster_path: string;
+  overview?: string;
+  budget: number;
+  revenue: number;
+  genres: string[];
+  runtime?: number;
+};
+
 export const MovieForm = ({ title, movie, closeCallback }: MovieFormProps) => {
-  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
-    e.preventDefault();
-    console.log("HandleSubmit");
-  }
+  const [backendErrors, setBackendErrors] = useState([]);
+  const [titleString, setTitleString] = useState(movie?.title);
+  const [releaseDate, setReleaseDate] = useState(movie?.release_date);
+  const [tagLine, setTagLine] = useState(movie?.tagline);
+  const [voteAverage, setVoteAverage] = useState(movie?.vote_average);
+  const [posterPath, setPosterPath] = useState(movie?.poster_path);
+  const [overview, setOverview] = useState(movie?.overview);
+  const [budget, setBudget] = useState(movie?.budget);
+  const [revenue, setRevenue] = useState(movie?.revenue);
+  const [genre, setGenre] = useState(movie?.genres);
+  const [runtime, setRuntime] = useState(movie?.runtime);
```

Comment:
The point of using form libraries (among other things) is that you DON'T handle each input manually. `movie` should only be used to define the default values for the form. See https://react-hook-form.com/get-started#IntegratingwithUIlibraries example for default values.

Reply:
fixed